src/integrations/ga.js

```diff
@@ -1,9 +1,15 @@
+import { generate } from '../guids.js';
+
 let pollAttempt = 0;
 const maxPollAttempts = 60;
 
 function gaIntegration() {
     if (pollAttempt > maxPollAttempts) {
-        console.log('Evolv: unable to set uid');
+        console.log('Evolv: GA clientId not found, using generated UID');
+        var id = generate();
+
+        window.evolv.setUid(id);
+        window.evolv.client.emit('evolv-uid.generated');
```

Comment:
Just checking if this is the desired order. Everything in the `main()` function and all of its events will emit before `evolv-uid.generated` is emitted.

Reply:
I think it's OK if the event fires after `main()` runs - `window.evolv.client.emit()` is not present until after `main()` starts to run again, and this is a secondary metric to gain some data points for when we have to use a generated uid. 
What are your thoughts?

Reply:
I didn't have an opinion because I don't know how the `evolv-uid.generated` event is used. I was just checking.

Reply:
Yeah, it's just a secondary metric I want to fire to track how often we are falling back to the generated UID